File: 804-Unique Morse Code Words-唯一摩尔斯密码词/804-Unique Morse Code Words-.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function(words) {
    var code = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    var codeArray = [];
    var uniqueArray = [];
    for(let word of words){
      var item = "";
      for(let i = 0 ;i<word.length;i++){
        item += code[word[i].charCodeAt()-97];
      }
      codeArray.push(item);
    }
    for(let item of codeArray){
        if(!uniqueArray.includes(item)){
          uniqueArray.push(item);
        }
      }
    return uniqueArray.length;
};

File: 804-Unique Morse Code Words-唯一摩尔斯密码词/804-Unique Morse Code Words-.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
      String code[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
	      ArrayList<String> codeString = new ArrayList<String>();
	      for (int j = 0; j<words.length;j++) {
	    	  String item = "";
	    	  char itemArray[] = words[j].toCharArray();
			for (int i = 0; i < itemArray.length; i++) {
				item +=code[Character.valueOf(itemArray[i])-97];
			}
			codeString.add(item);
		}
	      ArrayList<String> uniqueArrayList = new ArrayList<String>();
			for (String uniqItem : codeString) {
				if(!uniqueArrayList.contains(uniqItem)) {
					uniqueArrayList.add(uniqItem);
				}
			}
			return uniqueArrayList.size();
    }
}
